sql/updates/latest-dev.sql

```diff
@@ -63,3 +63,21 @@ CREATE INDEX continuous_aggs_materialization_invalidation_log_idx ON
     _timescaledb_catalog.continuous_aggs_materialization_invalidation_log (
         materialization_id, lowest_modified_value ASC);
 GRANT SELECT ON  _timescaledb_catalog.continuous_aggs_materialization_invalidation_log TO PUBLIC;
+
+-- Suspend any running retention policies that conflict with continuous aggs
+DO $$
+DECLARE
+  jobid INTEGER;
+BEGIN
+    FOR jobid IN
+        SELECT c.id
+        FROM _timescaledb_config.bgw_job a
+        LEFT JOIN _timescaledb_catalog.continuous_agg b ON a.hypertable_id = b.mat_hypertable_id
+        INNER JOIN _timescaledb_config.bgw_job c ON c.hypertable_id = b.raw_hypertable_id
+        WHERE a.proc_name = 'policy_refresh_continuous_aggregate' AND c.proc_name = 'policy_retention' AND c.scheduled
+            AND (a.config->'start_offset')::text::interval > (c.config->'drop_after')::text::interval
+    LOOP
+        RAISE NOTICE 'Retention job with id % will drop chunks while a continuous aggregate is still running on them. Suspending this job for now. If this is the desired behavior, you can reenable the job with: SELECT alter_job(%, scheduled=>true);', jobid, jobid;
```

Comment:
I'm not sure this notice will suggest to users the potential risks from re-enabling --> they might just blindly do so.  Maybe something like the following, although it is long and may be shortened.

```suggestion
        RAISE NOTICE 'Data retention policy with job id % will drop chunks while a continuous aggregate may refresh the dropped region. Suspending this job for now, as otherwise data in the continuous aggregate view will be replaced with results from the now-deleted region.  To adjust the behavior, alter the 'drop_after' parameter of your data retention policy.  You can re-enable the data retention policy with: SELECT alter_job(%, scheduled=>true);', jobid, jobid;
```

Reply:
I would prefer if we split this up in a simple main message and detail/hint as we do for normal errors/notices. Something like:

```
RAISE NOTICE 'suspending retention policy with ID %', jobid
      USING DETAIL = 'The retention policy was not running correctly so it was suspended. Enabling it again might remove data from the continuous aggregates.'
      USING HINT = 'Adjust the policy settings and enable the policy again with: SELECT alter_job(%, scheduled=>true).', jobid;
```

Please try to also follow the error/message style guide as any normal message.

Reply:
Hmm, I do like the idea of providing a more descriptive message here.  However we do encourage people to just just drop and recreate retention policies rather than try to update the jsonb config through alter_job.  Maybe this instead:
```        RAISE NOTICE 'Data retention policy (formerly drop_chunk policy) with job id % will drop chunks while a continuous aggregate is still running on them. Suspending this job as it will likely result in overwriting the aggregate with empty data. If this is actually the desired behavior, you can reenable the job with: SELECT alter_job(%, scheduled=>true);  Otherwise, please create a new rention_policy with a larger drop_after parameter and remove the old policy with: SELECT delete_job(%)', jobid, jobid, jobid;```

Reply:
Ahh, just saw Erik's comment, let me rework this a bit more.

Reply:
```        
RAISE NOTICE 'Suspending data retention policy with job id %.', jobid
            USING DETAIL = 'The retention policy (formerly drop_chunks policy) will drop chunks while a continuous aggregate is still running on them. This will likely result in overwriting the aggregate with empty data.',
            HINT = ('To restore the retention policy, with the possibility of updating aggregates with dropped data, run: SELECT alter_job(%, scheduled=>true);  Otherwise, please create a new rention_policy with a larger drop_after parameter and remove the old policy with: SELECT delete_job(%);', jobid, jobid);
 ```

Reply:
Note that the first letter of the main message should not be capitalized: https://www.postgresql.org/docs/current/error-style-guide.html

I don't think the hint should be in parentheses.

Reply:
Good catch, fixed